.github/workflows/compilers.yml

```diff
@@ -0,0 +1,201 @@
+name: Compilations
+
+on: [push, pull_request]
+
+# Github actions does not support YAML anchors.  This creative use of
+# environment variables (plus the "echo ::set-env" hack) is to reroute that
+# restriction.
+env:
+  default_cc: clang-11
+  append_cc: ''
+
+  # -O1 is faster than -O3 in our tests... Majority of time are consumed trying
+  # to optimize binaries.  Also Github Actions runs on a relatively modern CPU
+  # compared to, say, GCC 4 or Clang 3.  We don't specify `-march=native`
+  # because compilers tend not understand what the CPU is.
+  optflags: '-O1 -march=x86-64 -mtune=generic'
+
+  # -g0 disables backtraces when SEGV.  Do not set that.
+  debugflags: '-ggdb3'
+
+  default_configure: >-
+    --enable-debug-env
+    --disable-install-doc
+    --with-ext=-test-/cxxanyargs,+
+  append_configure: >-
+    --without-valgrind
+    --without-jemalloc
+    --without-gmp
+
+  UPDATE_UNICODE: >-
+    UNICODE_FILES=.
+    UNICODE_PROPERTY_FILES=.
+    UNICODE_AUXILIARY_FILES=.
+    UNICODE_EMOJI_FILES=.
+  CONFIGURE_TTY: never
+  RUBY_DEBUG: ci rgengc
+  RUBY_TESTOPTS: >-
+    -q
+    --color=always
+    --tty=no
+
+jobs:
+  compile:
+    strategy:
+      fail-fast: false
+      matrix:
+        entry:
+          - { key: default_cc, name: gcc-9,     value: gcc-9 }
+          - { key: default_cc, name: gcc-8,     value: gcc-8 }
+          - { key: default_cc, name: gcc-7,     value: gcc-7 }
+          - { key: default_cc, name: gcc-6,     value: gcc-6 }
+          - { key: default_cc, name: gcc-5,     value: gcc-5 }
+          - { key: default_cc, name: gcc-4.8,   value: gcc-4.8 }
+          - { key: default_cc, name: clang-11,  value: clang-11 }
+          - { key: default_cc, name: clang-10,  value: clang-10 }
+          - { key: default_cc, name: clang-9,   value: clang-9 }
+          - { key: default_cc, name: clang-8,   value: clang-8 }
+          - { key: default_cc, name: clang-7,   value: clang-7 }
+          - { key: default_cc, name: clang-6.0, value: clang-6.0 }
+          - { key: default_cc, name: clang-5.0, value: clang-5.0 }
+          - { key: default_cc, name: clang-4.0, value: clang-4.0 }
+          - { key: default_cc, name: clang-3.9, value: clang-3.9 }
+
+          - { key: append_cc, name: c99,   value: '-std=c99   -Werror=pedantic -pedantic-errors' }
+          - { key: append_cc, name: c11,   value: '-std=c11   -Werror=pedantic -pedantic-errors' }
+          - { key: append_cc, name: c17,   value: '-std=c17   -Werror=pedantic -pedantic-errors' }
+          - { key: append_cc, name: c2x,   value: '-std=c2x   -Werror=pedantic -pedantic-errors' }
+          - { key: CXXFLAGS,  name: c++98, value: '-std=c++98 -Werror=pedantic -pedantic-errors -Wno-c++11-long-long' }
```

Comment:
Do you remember the reason why you added this C++98 check? We've only supported C99 or later since https://bugs.ruby-lang.org/issues/15347, and I believe C++98 is not compatible with C99. In that policy, it'd be fair to say we only support C++11 or later, but is there any compiler that doesn't support C++11 but we still support?

Reply:
Each header file in this PR actually explained it (thanks, Alan for finding it). Never mind.

> We assume C99  for ruby itself but we don't  assume languages of
> extension libraries. They could be written in C++98.